contrib/!source-control/github/packages.el

```diff
@@ -86,3 +88,47 @@
         "gfC" 'spacemacs/git-link-commit-copy-url-only)
       ;; default is to open the generated link
       (setq git-link-open-in-browser t))))
+
+;; magit-gh-pulls has to be loaded via a pre-init hook because the source code
+;; makes assumptions about the status of the magit-mode keymaps that are
+;; incompatible with the spacemacs' evilification feature.
+;; To avoid errors, magit-gh-pulls must be loaded after magit, but before magit
+;; is configured by spacemacs. 
+
+(defun github/init-magit-gh-pulls ())
+
+(defun github/pre-init-magit ()
+  (spacemacs|use-package-add-hook magit
+    :pre-config
+    (progn
+      (use-package magit-gh-pulls
+        ;; only load this config for magit-gh-pulls if both 1 and 2 are true:
+        ;; 1) The package is used - this allows users to :exclude the package
+        ;; normally.
+        ;; 2) Magit-gh-pulls is owned by the github layer - this prevents this
+        ;; configuration from being run if another layer has claimed ownership
+        ;; of magit-gh-pulls, and maintains the spacemacs contract of a single
+        ;; owner per package.
+        :if (and (configuration-layer/package-usedp 'magit-gh-pulls)
+                 (equal (oref (-find (lambda (pkg) (equal (oref pkg :name)
+                                                          'magit-gh-pulls))
+                                     configuration-layer--packages)
+                              :owner)
+                        'github))
```

Comment:
Moving `use-package magit-gh-pulls` to its `init` function will work as well and its simpler, you just have to check that magit is a used package.

The time when you add the `:pre-config` hook to magit does not matter since magit `:config` is lazy loaded (the time would have been an important factor if you were adding a `:pre-init` or `post-init` hook since the `:init` block are evaluated right away at initialization time).